rulewerk-core/src/main/java/org/semanticweb/rulewerk/core/reasoner/implementation/FileDataSource.java

```diff
@@ -54,32 +55,33 @@
 	/**
 	 * Constructor.
 	 *
-	 * @param file               a file that will serve as storage for fact terms.
+	 * @param filePath           path to a file that will serve as storage for fact
+	 *                           terms.
 	 * @param possibleExtensions a list of extensions that the files could have
 	 * @throws IOException              if the path of the given {@code file} is
 	 *                                  invalid.
 	 * @throws IllegalArgumentException if the extension of the given {@code file}
 	 *                                  does not occur in
 	 *                                  {@code possibleExtensions}.
 	 */
-	public FileDataSource(@NonNull final File file, final Iterable<String> possibleExtensions) throws IOException {
-		Validate.notNull(file, "Data source file cannot be null!");
-		final String fileName = file.getName();
-
-		this.file = file;
+	public FileDataSource(final String filePath, final Iterable<String> possibleExtensions) throws IOException {
+		Validate.notBlank(filePath, "Data source file name cannot be null!");
+
+		this.file = new File(filePath);
+		this.filePath = filePath.replaceAll("\\\\", "/"); // canonicalise windows-style path separators
+		this.fileName = this.filePath.substring(this.filePath.lastIndexOf("/") + 1); // just the file name
```

Comment:
as `\\` is a valid string occurring in a Unix file name, fileName and filePath would lead to undesired behaviour. 
fileName should be obtained from the File object, and filePath that is returned by getPath() and used in the serialization should be unmodified, as it is user input.